src/platform_impl/linux/x11/mod.rs

```diff
@@ -547,7 +611,8 @@ impl<T: 'static> EventLoop<T> {
                     control_flow,
                     &mut |event, window_target, control_flow| {
                         if let Event::RedrawRequested(crate::window::WindowId(wid)) = event {
-                            wt.redraw_sender.send(wid).unwrap();
+                            wt.redraw_sender.sender.send(wid).unwrap();
+                            wt.redraw_sender.waker.ping();
```

Comment:
Could you wrap `send` in o `redraw_sender`, so it automatically `pings` instead of having comments around that you should ping and such?

Reply:
right - I was generally just re-instating code similar to how it was recently, except using a calloop Ping to wake.

I've now implemented `send()` for `WakeSender` and `EventLoopProxy` now also uses a `WakeSender` so the `waker.ping()` is now done automatically for redraw requests and sending user events.